tests/resource_monitor_plugin_test.py

```diff
@@ -112,6 +114,62 @@ def testCommon(title, extraNodeosArgs, expectedMsgs):
         if not isMsgInStderrFile(msg):
             errorExit ("Log should have contained \"%s\"" % (expectedMsgs))
 
+def extractTimestamp(msg):
+    matches = re.compile("\s+([0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}.[0-9]{3})\s").search(msg)
+    return datetime.strptime(matches.group(0).strip(), '%Y-%m-%dT%H:%M:%S.%f')
+
+intervalTolerance = 0.1 # 10%
```

Comment:
is 0.1 just random number? if yes could you run this test in a loop and dump to log ```curInterval - timedelta(seconds=expectedInterval)```
to see how volatile it could be?
This is to avoid another flakey test where we need restart tests 2-3 times to get it passed.

Reply:
0.1 (i.e. 10%) is based on observed warning interval deviations on buildkite and my local dockers. On buildkite, the max deviation is observed on macOS 10.15, where for the first test case (10-seconds interval, from 2 seconds monitor interval and warning interval 5) the deviation is about 0.6 sec, i.e. 6%. The second test case (30 seconds) has smaller deviation percentage. 
To be safer, I changed this tolerance to be 0.15 (15%) in the new commit.